### PR TITLE
used new version of nette/caching for support namespace; now may use …

### PR DESCRIPTION
**Pull request by namo-R**
New idea of using redis by namespaces

You are limited by numbers and by number of databases in redis. So idea is use namespaces for better management (New realease Nette/caching use namespaces too). I am not convinced about right of this pull request, but for me is this solution how better manage my saved data.

### CODE REVIEW

File: src/Kdyby/Redis/DI/RedisExtension.php [hunk 1]
```diff
@@ -52,6 +52,7 @@ class RedisExtension extends Nette\DI\CompilerExtension
 		'lockAcquireTimeout' => FALSE,
 		'debugger' => '%debugMode%',
 		'versionCheck' => TRUE,
+		'namespace' => ""
```
Comment (enumag) [inline src/Kdyby/Redis/DI/RedisExtension.php]: Use single quotes.

File: src/Kdyby/Redis/DI/RedisExtension.php [hunk 2]
```diff
@@ -160,8 +161,12 @@ protected function loadJournal(array $config)
 		$builder->removeDefinition($journalService);
 		$builder->addDefinition($journalService)->setFactory($this->prefix('@cacheJournal'));
 
+		$namespace = NULL;
+		if(isset($config['namespace']))
+			$namespace = $config['namespace'];
```
Comment (enumag) [inline src/Kdyby/Redis/DI/RedisExtension.php]: `$namespace = $config['namespace'] ?? null;`

File: src/Kdyby/Redis/DI/RedisExtension.php [hunk 3]
```diff
@@ -160,8 +161,12 @@ protected function loadJournal(array $config)
...
+
 		$builder->addDefinition($this->prefix('cacheJournal'))
-			->setClass('Kdyby\Redis\RedisLuaJournal');
+			->setClass('Kdyby\Redis\RedisLuaJournal', [$builder->getDefinition('redis.client'), $namespace]);
```
Comment (enumag) [inline src/Kdyby/Redis/DI/RedisExtension.php]: Use `setArguments` method instead.

File: src/Kdyby/Redis/RedisJournal.php [hunk 4]
```diff
@@ -37,14 +37,21 @@ class RedisJournal implements Nette\Caching\Storages\IJournal
...
+	protected $namespace = "";
+
 
 
 	/**
 	 * @param RedisClient $client
+	 * @param [string] $namespace
 	 */
-	public function __construct(RedisClient $client)
+	public function __construct(RedisClient $client, $namespace = "")
```
Comment (enumag) [inline src/Kdyby/Redis/RedisJournal.php]: single quotes

File: src/Kdyby/Redis/RedisJournal.php [hunk 5]
```diff
@@ -37,14 +37,21 @@ class RedisJournal implements Nette\Caching\Storages\IJournal
...
+	/**
+	 * Namespace for journal
+	 */
+	protected $namespace = "";
+
 
 
 	/**
 	 * @param RedisClient $client
+	 * @param [string] $namespace
```
Comment (enumag) [inline src/Kdyby/Redis/RedisJournal.php]: `@param string $namespace`

**Comment by enumag** [comment]
I'm not quite sure about this either. I need opinions from other people about this.

**Comment by jkavalik** [comment]
@enumag What is the status of this please?

**Comment by enumag** [comment]
@jkavalik Still the same. I'm not willing to merge this without opinions of other Kdyby maintainers or contributors.

**Comment by namo-R** [comment]
Branch recreated, I made a mistake. Can you reopen ?

**Comment by enumag** [comment]
It's not possible unfortunately.

**Pull request closed without merging**